Source/Core/DolphinTool/HeaderCommand.cpp

```diff
@@ -52,19 +57,58 @@ int HeaderCommand(const std::vector<std::string>& args)
     return EXIT_FAILURE;
   }
 
+  const bool enable_json = options.is_set_by_user("json");
   const bool enable_block_size = options.is_set_by_user("block_size");
   const bool enable_compression_method = options.is_set_by_user("compression");
   const bool enable_compression_level = options.is_set_by_user("compression_level");
 
-  // Open the blob reader, plus get blob type
+  // Open the blob reader
   const std::unique_ptr<DiscIO::BlobReader> blob_reader = DiscIO::CreateBlobReader(input_file_path);
   if (!blob_reader)
   {
     fmt::print(std::cerr, "Error: Unable to open disc image\n");
     return EXIT_FAILURE;
   }
+  // Open the volume
+  const std::unique_ptr<DiscIO::Volume> volume = DiscIO::CreateVolume(blob_reader->CopyReader());
+
+  if (enable_json)
+  {
+    auto json = picojson::object();
+
+    // File data
+    if (const u64 block_size = blob_reader->GetBlockSize())
+      json["block_size"] = picojson::value((double)block_size);
+
+    const std::string compression_method = blob_reader->GetCompressionMethod();
+    if (compression_method != "")
+      json["compression_method"] = picojson::value(compression_method);
+
+    if (const std::optional<int> compression_level = blob_reader->GetCompressionLevel())
+      json["compression_level"] = picojson::value((double)compression_level.value());
+
+    // Game data
+    if (volume)
+    {
+      json["name"] = picojson::value(volume->GetInternalName());
```

Comment:
I would recommend calling this `internal_name` instead of `name`. This name is frequently unreliable, and apps typically shouldn't use it to identify games to the user unless there is no other option.

Reply:
Heh, I originally called it that then changed it when I realized the GUI just called it Name in the Properties -> Info panel. I'll change it back 👍